terminus-ui/card/src/card.component.spec.ts

```diff
@@ -1,95 +1,139 @@
+  `,
+})
+class TestHostComponent {
+  disabled!: boolean;
+  flat!: boolean;
+  supportsInteraction!: boolean;
+
+  @ViewChild(TsCardComponent)
+  cardComponent!: TsCardComponent;
+}
+
+describe(`TsCardComponent`, () => {
+  let component: TestHostComponent;
+  let fixture: ComponentFixture<TestHostComponent>;
+  let card: HTMLElement;
+  let cardComponent: TsCardComponent;
+  let cardComponent2: TsCardComponent;
 
   beforeEach(() => {
-    component = new TsCardComponent();
-    component2 = new TsCardComponent();
+    fixture = createComponent(TestHostComponent, [], [TsCardModule]);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+    cardComponent = component.cardComponent;
+    card = fixture.debugElement.query(By.css('.c-card')).nativeElement as HTMLElement;
+    cardComponent2 = new TsCardComponent();
   });
 
+  describe(`isDisabled`, () => {
+    test(`card is not disabled`, () => {
+      expect(cardComponent.disabled).toEqual(false);
+      component.disabled = false;
+      fixture.detectChanges();
+      expect(cardComponent.isDisabled).toEqual(false);
+      expect(cardComponent.disabled).toEqual(false);
+      expect(card.classList).not.toContain('c-card--disabled');
+    });
 
-  it(`should exist`, () => {
-    expect(component).toBeTruthy();
+    test(`card is disabled`, () => {
+      component.disabled = true;
+      fixture.detectChanges();
+      expect(cardComponent.isDisabled).toEqual(true);
+      expect(cardComponent.disabled).toEqual(true);
+      expect(card.classList).toContain('c-card--disabled');
+    });
   });
 
-  describe(`centeredContent`, () => {
-    it(`should set and retrieve`, () => {
-      component.centeredContent = true;
-      expect(component.centeredContent).toEqual(true);
+  describe(`isFlat`, () => {
+    test(`card is flat`, () => {
+      component.flat = true;
+      fixture.detectChanges();
+      expect(cardComponent.flat).toEqual(true);
+      expect(card.classList).toContain('c-card--flat');
+    });
+    test(`card is not flat`, () => {
+      component.flat = false;
+      fixture.detectChanges();
+      expect(cardComponent.flat).toEqual(false);
+      expect(card.classList).not.toContain('c-card--flat');
     });
   });
 
   describe(`supportsInteraction`, () => {
-    test(`should set and retrieve`, () => {
+    test(`card supports interaction`, () => {
       component.supportsInteraction = true;
-      expect(component.supportsInteraction).toEqual(true);
+      fixture.detectChanges();
+      expect(cardComponent.supportsInteraction).toEqual(true);
+      expect(card.classList).toContain('c-card--interaction');
+    });
+    test(`card does not support interaction`, () => {
+      component.supportsInteraction = false;
+      fixture.detectChanges();
+      expect(cardComponent.supportsInteraction).toEqual(false);
+      expect(card.classList).not.toContain('c-card--interaction');
     });
   });
 
+  describe(`centeredContent`, () => {
```

Comment:
I wonder if there is a better way to test if this is actually happening.. What do you think of something like this:

Test component setup:
```html
<ts-card [centeredContent]="true" id="outer">
  <!-- inline element so that the container doesn't go full width -->
  <span id="inner">
    Centered content
  </span>
</ts-card>
```

Test:

```typescript
// Get the dimensions and position of both elements
const outerDimensions = document.getElementById('outer').getBoundingClientRect();
const innerDimensions = document.getElementById('inner').getBoundingClientRect();
// Find the space between the right edge of the inner to the right edge of the outer
const spaceOnRight = outerDimensions.right - innerDimensions.right;
const spaceOnLeft = innerDimensions.left;
// Find the difference between the space on either side and force to a positive number
const value = Math.abs(spaceOnLeft - spaceOnRight);

// We give it a buffer of 50 in case it's not exact
expect(value).toBeLessThan(50);
```

Thoughts?

Reply:
Interesting. I'll play with it for a bit. It's probably easier to test `centeredContent` being true than being false.

Reply:
Yeah I had to use it in a few places for the select panel math